src/monty/json.py

```diff
@@ -586,8 +566,8 @@ def default(self, o) -> dict:
         if isinstance(o, Path):
             return {"@module": "pathlib", "@class": "Path", "string": str(o)}
 
-        if torch is not None and isinstance(o, torch.Tensor):
-            # Support for Pytorch Tensors.
+        # Support for Pytorch Tensors
+        if _check_type(o, "torch.Tensor"):
```

Comment:
Should be the correct type string:
```
import torch


tensor = torch.randn(3, 3)
print(type(tensor).mro())
```

Gives: `[<class 'torch.Tensor'>, <class 'torch._C.TensorBase'>, <class 'object'>]`